### ISSUE TITLE
Enhance `burn-wgpu` for WebGL Fallback Support

### ISSUE THREAD

**Issue by antimora** [tracel-ai/burn#852]
**Description:**
As we aim to make `burn-wgpu` more versatile in terms of browser compatibility, there's a need to enhance its ability to fall back to WebGL when WebGPU is unsupported.

**Details:**
- Investigate the capability of the `wgpu` project in terms of supporting WebGL. An example with the `webgl` feature flag has been observed [here](https://github.com/gfx-rs/wgpu#rust).
  
**Advantages:**
- Enhanced browser compatibility: As of now, WebGPU support is limited to Chrome 113 on both Mac and Windows. By having a fallback mechanism to WebGL, we can ensure that our application runs seamlessly across a broader range of browsers.

**Next Steps:**
1. Review the mentioned `wgpu` example with the `webgl` feature flag.
2. Experiment with implementing a WebGL fallback in a local branch.
3. Test on browsers without WebGPU support and document results. Use [image-classfication-web](https://github.com/burn-rs/burn/tree/main/examples/image-classification-web) example.
4. Merge changes to the main branch upon successful testing.

**Additional Resources:**
- [wgpu with webgl feature flag](https://github.com/gfx-rs/wgpu#rust)

**Comment by DavidGOrtega** [comment]
First of all, thanks a lot for all the effort put into burn, its amazing.

Regarding to this issue. My personal opinion is that the project should focus in just WGPU. I had some experience working with WebGL for tensorflowjs in the past and it might not be worth, working with shaders and GLSL is a very unpleasant experience. To me is adding a separate backend effort which might not be worth.

Hope this helps 🙇

**Comment by antimora** [comment]
@DavidGOrtega i agree we should not add separate WebGL kernels. The scope of this issue is to explore an easier solution that relies on wgpu shader translation mechanisms. If there is no such solution, we can consider this issue as closed.
